src/index.ts

```diff
@@ -1,17 +1,53 @@
+const fs = require('fs');
+
 const core = require('@actions/core');
 const exec = require('@actions/exec');
 
-import buildExec from './buildExec';
+const superagent = require('superagent');
+
+// import buildExec from './buildExec';
+
+// const {failCi} = buildExec();
+
+try {
+  const uploader = () => {
+    return async () => {
+      try {
+        const uploadBinary = await superagent.get('https://uploader.codecov.io/latest/codecov-linux');
+        return uploadBinary;
+      } catch (err) {
+        core.setFailed(
+            'Codecov: Could not properly download uploader binary: ' +
+            `${err.message}`,
+        );
+      }
+    };
+  };
+  console.log(uploader);
+  const filename = __dirname + '/uploader';
+  fs.writeFileSync(filename, uploader);
+  console.log('wrote it');
+
+  fs.chmodSync(filename, '700');
 
-const {execArgs, options, failCi} = buildExec();
+  console.log('Did it');
+  console.log(fs.readdirSync(__dirname));
+  console.log(__dirname);
 
-exec.exec('bash', execArgs, options)
-    .catch((err) => {
-      if (failCi) {
+  () => {
+    return async () => {
+      try {
+        await exec.exec(filename);
+      } catch (err) {
         core.setFailed(
-            `Codecov failed with the following error: ${err.message}`,
+            'Codecov: Could not properly run uploader binary: ' +
+            `${err.message}`,
         );
-      } else {
-        core.warning(`Codecov warning: ${err.message}`);
       }
-    });
+    };
+  };
```

Comment:
This does not run. The expression evaluates to `undefined`. At a high level, what actually is parsed and evaluated is similar to the following:

```ts
() => {
  // …
};
```

The kind of pattern that was most likely intended is called an [immediately-invoked function expression (IIFE)](https://developer.mozilla.org/en-US/docs/Glossary/IIFE). IIFEs take the pattern similar to the following:

```ts
// (1)
(function () {
  // …
})();

// (2)
(() => {
  // …
})();
```

Taking a step back, these are all different ways of writing a function expression wherein the function being called is already defined:

```ts
function greet() {
  return "Hello, world!";
}

greet();
// => "Hello, world!"

(greet)();
// => "Hello, world!"
```

I implore you to play around with this functionality in the Node.js REPL:

```console
$ node
Welcome to Node.js v16.4.2.
Type ".help" for more information.
> 
```

Reply:
Thanks for the great write-up here. I was fed up last night and took a break for the night